reference/ast-examples.md

```diff
@@ -0,0 +1,185 @@
+# AST Examples
+
+When building recipes, it's important to understand how OpenRewrite Abstract Syntax Trees (AST) correspond to code. You couldn't, for example, properly rename a variable with a recipe unless you knew that [J.Identifier](https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/java/org/openrewrite/java/tree/J.java#L2310-L2343) is the class used to represent a variable.
+
+To help you get started on working with ASTs and OpenRewrite, this guide will:
```

Comment:
We should mention that this is a guide to the Java AST. XML, Protobuf, Yaml, etc., each have their own AST. Being data formats those are generally simpler and more approachable structures.